backend/layers/persistence/persistence_mock.py

```diff
@@ -452,6 +452,15 @@ def get_all_versions_for_dataset(self, dataset_id: DatasetId) -> List[DatasetVer
                 versions.append(self._update_dataset_version_with_canonical(dataset_version))
         return versions
 
+    def check_artifact_is_part_of_dataset(self, dataset_version_id: DatasetVersionId, artifact_id: DatasetArtifactId):
+        dataset_version = self.datasets_versions[dataset_version_id.id]
+        return any(artifact.id == artifact_id for artifact in dataset_version.artifacts)
+
+    def get_artifact_by_uri_suffix(self, uri_suffix: str) -> Optional[DatasetArtifact]:
+        for artifact in self.dataset_artifacts.values():
+            if artifact.uri.endswith(uri_suffix):
+                return artifact
+
```

Comment:
I don't think either of these methods are called anywhere

Reply:
Same goes for the other definitions of these methods too

Reply:
they are called when we run the backend tests not in integration mode

Reply:
Can you link to that? When I search the your branch for usage I don't see it:

![image](https://github.com/user-attachments/assets/76bb9d5e-121f-4fff-9215-42ecba8dc4e3)

![image](https://github.com/user-attachments/assets/64cb120b-0934-4cee-8095-a3012308e1aa)

Reply:
there are a few places where the unit tests fork and either use `DatabaseProviderMock`, or `DatabaseProvider`,  for example here https://github.com/chanzuckerberg/single-cell-data-portal/blob/7cfda1a5a0f5bf272cdd0862e220041e57cdd6e7/tests/unit/backend/layers/common/base_test.py#L108-L108.

Reply:
Oh sorry for the confusion. 

My point here is not about the class. It is that I do not see these methods (the implementation here, or from any other class) being called anywhere. Like, if I do a search of the full codebase for the string `"check_artifact_is_part_of_dataset"` only the definitions show up. So these methods are currently dead code branches.

Reply:
Oh, you're right, they are part of my next PR